### PR TITLE
Refactor API deprecation, fix request constructors with positional args

### PR DESCRIPTION
**Pull request by wltsmrz**

### CODE REVIEW

File: src/core/remote.js [hunk 1]
```diff
@@ -767,6 +767,31 @@ Remote.prototype.request = function(request_) {
 };
 
 /**
+ *  Create options object from positional function arguments
+ *
+ * @param {Array} params function parameters
```
Comment (sublimator) [inline src/core/remote.js]: I like this form: `@param {Array} params - function parameters`.  Using a `-` after the param name or whatever provides some separation.

File: test/remote-test.js [hunk 2]
```diff
@@ -1873,3 +1960,499 @@ describe('Remote', function() {
     });
   });
 });
+
+describe.skip('Request API consistency tests', function() {
```
Comment (sublimator) [inline test/remote-test.js]: All these tests are skipped ?

File: test/remote-test.js [hunk 3]
```diff
@@ -1873,3 +1960,499 @@ describe('Remote', function() {
...
+  );
+  assert.deepEqual(
+    firstMethodParam(['account', ADDRESS]),
+    ['account', ADDRESS]
+  );
+
+  // Filters default params against optional params
+  function filterDefault(optionalArgs, defaultArg) {
+    return lodash.isEmpty(defaultArg)
+    || lodash.isEmpty(optionalArgs)
```
Comment (sublimator) [inline test/remote-test.js]: I find it easier to read multi-line expression when lines after the first are indented.

File: test/remote-test.js [hunk 4]
```diff
@@ -1873,3 +1960,499 @@ describe('Remote', function() {
...
+      ]
+    }),
+    makeRequest('submit', 'requestSubmit')
+  ];
+
+  function checkRequest(request, expectedReqParams) {
+    assert(request.requested, 'Request unattempted, most likely callback was ignored');
+    [['id', undefined], ...expectedReqParams].forEach(arg => {
+      assert.deepEqual(
+        request.message[arg[0]], arg[1],
```
Comment (sublimator) [inline test/remote-test.js]: I `skipped the skip` and ran these tests locally and it seems there's some tests where `arg[0]` is not a string (key), but actually a [key, value] list it self.

File: src/core/remote.js [hunk 5]
```diff
@@ -1660,23 +1646,21 @@ Remote.prototype.requestBookOffers = function(options_, callback_) {
...
-    log.warn('DEPRECATED: First argument to request constructor should be'
-      + ' an object containing request properties');
-
-    options = {
-      seed: arguments[0]
-    };
+    lodash.merge(options, makeOptions(
+      'wallet_accounts',
+      ['seed'],
+      Array.prototype.slice.call(arguments)
```
Comment (sublimator) [inline src/core/remote.js]: https://lodash.com/docs#slice ?

File: src/core/remote.js [hunk 6]
```diff
@@ -1209,35 +1212,21 @@ Remote.prototype.requestTx = function(options_, callback_) {
...
-    options = {
-      account: args.shift(),
-      peer: args.shift(),
-      ledger: args.shift(),
-      limit: args.shift(),
-      marker: args.shift()
-    };
+    lodash.merge(options, makeOptions(
+      command,
+      ['account', 'ledger', 'peer', 'limit', 'marker'],
```
Comment (sublimator) [inline src/core/remote.js]: Was this change intentional?  It went account/peer/ledger before and now it's account/ledger/peer ?

File: src/core/remote.js (hunk 1 above)
Comment (wltsmrz) [inline src/core/remote.js]: Agree

File: test/remote-test.js (hunk 2 above)
Comment (wltsmrz) [inline test/remote-test.js]: Yeah. I'll unskip them when I've generated fixtures. Not sure if they're actually passing ATM. By the way, there's a bug with eslint that prohibits spread operator

Comment (wltsmrz) [inline test/remote-test.js]: Or I can just remove them if they don't seem useful

File: test/remote-test.js (hunk 3 above)
Comment (wltsmrz) [inline test/remote-test.js]: Agree

File: test/remote-test.js (hunk 4 above)
Comment (wltsmrz) [inline test/remote-test.js]: They were working the other day, maybe related to some linting I had to do for eslint bug re: spread operator? I'll fix them after I've generated fixtures, or just remove them now!

File: test/remote-test.js (hunk 2 above)
Comment (sublimator) [inline test/remote-test.js]: Yeah, maybe just remove them and add them when they are ready (using fixtures or whatever you wanted to do)?

File: src/core/remote.js (hunk 5 above)
Comment (wltsmrz) [inline src/core/remote.js]: Crossed my mind when I was writing this, thought lodash had a slice thing. I'll use that instead

File: src/core/remote.js (hunk 6 above)
Comment (wltsmrz) [inline src/core/remote.js]: Yeah

File: src/core/remote.js [hunk 7]
```diff
@@ -1336,7 +1322,20 @@ Remote.prototype.requestAccountCurrencies = function(...args) {
 Remote.prototype.requestAccountLines = function(...args) {
   // XXX Does this require the server to be trusted?
   // utils.assert(this.trusted);
-  const options = ['account_lines', ...args];
+  let options = ['account_lines'];
+
+  if (lodash.isPlainObject(args[0])) {
+    options = options.concat(args);
+  } else {
+    const [account, peer, ledger] = args;
```
Comment (sublimator) [inline src/core/remote.js]: Ah, I can see the reflection here

File: test/remote-test.js (hunk 2 above)
Comment (sublimator) [inline test/remote-test.js]: Do we have an issue open with eslint about spreads? Are we using the latest version of eslint ?

**Comment by sublimator** [comment]
LGTM
